inlang/source-code/sdk-v2/src/types/schema.ts

```diff
@@ -1,113 +1,127 @@
-import type { Insertable, JSONColumnType, Selectable, Updateable } from "kysely"
-
+import { Type, type Static } from "@sinclair/typebox"
+import type { Insertable, Selectable, Updateable } from "kysely"
 import type { LintReport } from "./lint.js"
 
-export type Database = {
-	bundle: BundleTable
-	message: MessageTable
-	variant: VariantTable
-
-	lintReport: LintReport
-	// todo - move out of database
-	settings: Settings
-}
-
-type AliasMap = {
-	[key: string]: string
-}
-
-// Bundles all languages of a message -
-type BundleTable = {
-	id: string
-	// todo make alias relational
-	alias: JSONColumnType<AliasMap> // alias usually have a property "default"  that represents the message name like "welcome_message" or "login_button"
-	// messages[] @relation
-	// messages: Message[]
+export const VariableReference = Type.Object({
+	type: Type.Literal("variable"),
+	name: Type.String(),
+})
+
+export const Literal = Type.Object({
+	type: Type.Literal("literal"),
+	name: Type.String(),
+})
+
+export const Option = Type.Object({
+	name: Type.String(),
+	value: Type.Union([Literal, VariableReference]),
+})
+
+const FunctionAnnotation = Type.Object({
+	type: Type.Literal("function"),
+	name: Type.String(),
+	options: Type.Array(Option),
+})
+
+export const Expression = Type.Object({
+	type: Type.Literal("expression"),
+	arg: Type.Union([VariableReference, Literal]),
+	annotation: Type.Optional(FunctionAnnotation),
+})
+
+export const Text = Type.Object({
+	type: Type.Literal("text"),
+	value: Type.String(),
+})
+
+export const Declaration = Type.Object({
+	type: Type.Literal("input"),
+	name: Type.String(),
+	value: Expression,
+})
+
+export const Pattern = Type.Array(Type.Union([Text, Expression]))
+
+export const Settings = Type.Object({
+	baseLocale: Type.String(),
+	locales: Type.String(),
+	modules: Type.String(),
+})
+
+export const BundleTable = Type.Object({
+	id: Type.String(),
+	alias: Type.Any(), // TODO: Use appropriate Typebox type for JSONColumnType<AliasMap>
+})
+
+export const MessageTable = Type.Object({
+	id: Type.String(),
+	bundleId: Type.String(),
+	locale: Type.String(),
+	declarations: Type.Any(), // TODO: Use appropriate Typebox type for JSONColumnType<Declaration[]>
+	selectors: Type.Any(), // TODO: Use appropriate Typebox type for JSONColumnType<Expression[]>
```

Comment:
@samuelstroschein / @martin-lysk what are we going to do with the kysely types in typebox?

Reply:
@felixhaeberle for now we will just expose the sql db and do no validation on data we pass into the db - this should change and everything you update/insert should go through a runtime validation check.